src/services/MakeStreamZip.php

```diff
@@ -138,13 +145,23 @@ private function addTimestampFiles(int $id): void
      * @return void
      */
     private function nameFolder(): void
+    {
+        $this->folder = $this->getBaseFileName();
+    }
+
+    /**
+     * Folder and zip file name begins with date for experiments
+     *
+     * @return void
+     */
+    private function getBaseFileName(): string
     {
         if ($this->Entity instanceof Experiments) {
-            $this->folder = $this->Entity->entityData['date'] . ' - ' . Filter::forFilesystem($this->Entity->entityData['title']);
+            return $this->Entity->entityData['date'] . ' - ' . Filter::forFilesystem($this->Entity->entityData['title']);
         } elseif ($this->Entity instanceof Database) {
-            $this->folder = $this->Entity->entityData['category'] . ' - ' . Filter::forFilesystem($this->Entity->entityData['title']);
+            return $this->Entity->entityData['category'] . ' - ' . Filter::forFilesystem($this->Entity->entityData['title']);
         }
```

Comment:
either throw an exception at the end (like ImproperActionException('bad entity instance') or remove the elseif because the return makes it not necessary

Reply:
I think the Exception is more elegant as it is safer in case of future refactorings. I will change it as suggested.